fs/bcachefs/str_hash.h

```diff
@@ -46,18 +46,29 @@ bch2_hash_info_init(struct bch_fs *c, const struct bch_inode_unpacked *bi)
 	struct bch_hash_info info = {
 		.type = (bi->bi_flags >> INODE_STR_HASH_OFFSET) &
 			~(~0U << INODE_STR_HASH_BITS),
-		.crc_key = bi->bi_hash_seed,
```

Comment:
There is definitely a way to fix this in less lines, but I tried to break it out and pick the most readable fix. If we'd like to condense this some or if we don't want the BUG, I totally understand.